complete rewrite of github PR comments (#1293)
semgrep/semgrep-docs: docs/semgrep-cloud-platform/github-pr-comments.md

Semgrep can create **pull request (PR) comments** in your GitHub repository. These comments inform your developers of **findings**, such as security issues, in their PRs. Semgrep can also provide remediation tips or code fixes that your developers can click to **commit** into their code directly.

:::info Prerequisites
- Pull request (PR) comments can only be set up through Semgrep Cloud Platform (SCP). [<i class="fas fa-external-link fa-xs"></i> Create an account](/semgrep-code/getting-started/#signing-in-to-semgrep-cloud-platform) to set up PR comments.
- You must connect your GitHub organization (org) to Semgrep.
    - For **GitHub Cloud**: See [<i class="fa-regular fa-file-lines"></i> Adding an organization](/semgrep-cloud-platform/getting-started/#adding-an-organization).
    - For **GitHub Enterprise Server**: [<i class="fa-regular fa-file-lines"></i> Integrate Semgrep with GitHub Enterprise](/semgrep-cloud-platform/scm/#integrating-semgrep-cloud-platform-with-github-enterprise-or-gitlab-self-managed) and if you have a VPN or firewall, [set up your allowlists](/semgrep-cloud-platform/scm/#receiving-pr-or-mr-comments-in-your-vpn-or-on-premise-scm).

> - You must add or onboard a Project (repository) to SCP and it must complete at least one full scan successfully.

Reviewer: Why is project capitalized?

I also ran into a wording issue with the full scan thing; the way this is currently worded makes it seem like the project has to complete the full scan -- maybe something like "you must have scanned it fully at least once with Semgrep?"

Author: 1. Project is capitalized so they know it's a (Semgrep) Project - do you think we should continue this convention? (Happy to open a slack thread if necessary, otherwise we can just decide here).
2. Semgrep performs 2 kinds of scans, a "full scan" vs a diff-aware one. Both scans can actually be performed on unintended targets (full scan on PRs, diff-aware on main, we don't want people to do this, but it can happen.) We could also say:
- **Full scan on your default branch**
- Maybe: "your initial scan", because initial scans after adding to a project are full scans.
- 
What do you think?

Reviewer: Not to go off the other deep end, but let's stick with Semgrep Project so that it is crystal clear?

I think both "full scan on your default branch" or "your initial scan" works -- I would lean toward the former, just in case someone is setting this up after they've done multiple scans

Author: Resolved based on convo in slack!

:::
## Conditions for PR comment creation

> PR comments appear for the following types of scans and under certain conditions:

Reviewer: ```suggestion
PR comments appear for the following types of scans under these conditions:
```

<CommentTriggers />

Automated comments on GitHub pull requests are displayed as follows:

![Screenshot of a GitHub PR comment](/img/semgrep-pull-request.png#bordered)
**Figure** An inline GitHub pull request comment.

## Confirm account connection and access

### Confirm your Semgrep account's connection to GitHub

Confirm that you have the correct connection and access:

1. In your SCP account, click **Settings > Source code managers**.

> 2. Check that an entry for your GitHub org exists and is correct.

Reviewer: do we want to avoid using org the way we avoid repo?

Author: Fortunately, it is spared from an entry in the [style guide word list](https://developers.google.com/style/word-list). However, I do open with "organization (org)" in both GitHub (line 54) and GitLab (line 33).

1. Optional: Use the filters to quickly find rules to set to Comment or Block.
1. Click the **<i class="fa-solid fa-square-check"></i> checkbox** of the rules you want to set. You can use <kbd>Ctrl + Click</kbd> to select rules in bulk.
1. Click **Change modes**.
2. Click either **Block** or **Comment**.

You have successfully configured PR comments for Semgrep Code. If you are using **GitHub Actions** to run Semgrep, no extra changes are needed to receive PR comments.

### Enable autofix in GitHub repositories

[Autofix](/writing-rules/autofix) is a Semgrep feature in which rules contain suggested fixes to resolve findings.

<EnableAutofix />

### Dataflow traces in PR comments

![Screenshot of a GitHub PR comment with dataflow traces](/img/dataflow-traces-pr-comments.png#bordered)
**Figure** An inline GitHub pull request comment with dataflow traces.

<DisplayTaintedDataIntro />

#### View the path of tainted data in PR comments

To enable dataflow traces feature in your CI pipeline, fulfill the following prerequisites:

:::info Prerequisites

> - Set up GitHub pull request Semgrep comments, as described on this page.

Reviewer: ```suggestion
- Set up Semgrep to submit pull request comments to GitHub as described on this page.
```
